### PR TITLE
Fixes #29177 - Fix Style/EachWithObject cop

### PR DESCRIPTION
**Pull request by mmoll**

### CODE REVIEW

File: modules/dhcp_common/server.rb [hunk 1]
```diff
@@ -185,7 +185,7 @@ def clean_up_add_record_parameters(in_options)
       subnet = options.delete("network")
 
       # convert string keys to symbols
-      options = options.inject({}){|all, current| all[current[0].to_sym] = current[1]; all}
+      options = options.each_with_object({}){|current, all| all[current[0].to_sym] = current[1]; }
```
Comment (ekohl) [inline modules/dhcp_common/server.rb]:
```suggestion
      options = options.each_with_object({}){|current, all| all[current[0].to_sym] = current[1] }
```

File: modules/dhcp_native_ms/dhcp_native_ms_main.rb [hunk 2]
```diff
@@ -288,7 +286,7 @@ def sunw_option(option_id)
     end
 
     def generate_sunw_options_by_id
-      SUNW.inject({}) { |all, current| all[current[1][:code]] = current[0]; all }
+      SUNW.each_with_object({}) { |current, all| all[current[1][:code]] = current[0]; }
```
Comment (ekohl) [inline modules/dhcp_native_ms/dhcp_native_ms_main.rb]:
```suggestion
      SUNW.each_with_object({}) { |current, all| all[current[1][:code]] = current[0] }
```

File: test/plugins/plugin_initializer_test.rb [hunk 3]
```diff
@@ -40,7 +40,7 @@ def test_initialize_plugins
     assert all_but_uninitialized.all? {|p| p.has_key?(:di_container)}
 
     # filter out :di_container, can't use equality test with it
-    loaded = plugins.loaded.map {|p| [:name, :version, :class, :state, :http_enabled, :capabilities, :settings, :https_enabled].inject({}) {|a, c| a[c] = p[c]; a}}
+    loaded = plugins.loaded.map {|p| [:name, :version, :class, :state, :http_enabled, :capabilities, :settings, :https_enabled].each_with_object({}) {|c, a| a[c] = p[c]; }}
```
Comment (ekohl) [inline test/plugins/plugin_initializer_test.rb]:
```suggestion
    loaded = plugins.loaded.map {|p| [:name, :version, :class, :state, :http_enabled, :capabilities, :settings, :https_enabled].each_with_object({}) {|c, a| a[c] = p[c] }}
```

**Comment by mmoll** [comment]
There's a semicolon cop...

**Comment by ekohl** [comment]
I'm kind of fan of keeping these changes small and not touch a line too much.

File: modules/dhcp_common/server.rb (hunk 1 above)
Comment (lzap) [reply modules/dhcp_common/server.rb]: Wait, itsnt this a change? It returned `all`, now it returns what?

Comment (mmoll) [reply modules/dhcp_common/server.rb]: https://www.rubocop.org/en/stable/cops_style/#styleeachwithobject describes exactly this: "...without the need to return the object at the end."

**Comment by mmoll** [comment]
Unfortunately, neither Rubocop nor Rubymine supports an auto-fix for semicolons of these one-liners. Anyway, should all be good now.

**Review by ekohl — APPROVED** [review]

**Comment by ekohl** [comment]
Leaving it open so @lzap can have another look.

**Review by lzap — APPROVED** [review]
Haven't spotted anything wrong, thanks.

**Pull request merged**